tests/func/test_glr_parsing.py

```diff
@@ -275,7 +284,44 @@ def test_non_eof_grammar_empty():
     p = GLRParser(g_empty, debug=True)
 
     results = p.parse(txt)
-    assert len(results) == 3
+    assert len(results) == 8
 
     results = p.parse("")
     assert len(results) == 1
+
+
+def test_terminal_collision():
+    g = Grammar.from_string("""
+    expression: "1" s letter EOF
+              | "2" s "A" EOF
+              ;
+
+    s: " ";
+
+    terminals
+    letter: /[A-Z]/;
+    """)
+
+    p = GLRParser(g, ws='')
+
+    p.parse("2 A")
+    p.parse("1 B")
+    p.parse("1 A")
+
+
+def test_lexical_ambiguity():
+    g = Grammar.from_string("""
+    expression: a "x" EOF
+              | b EOF
+              ;
+
+    a: "x";
+    b: "xx";
+    """)
+
+    p = GLRParser(g)
+
+    assert sorted(p.parse("xx")) == [
+        ['x', 'x', None],
+        ['xx', None],
+    ]
```

Comment:
For testing of `lexical_disambiguation` above two tests could be extended with asserted results in case disambiguation is turned on.

Reply:
I didn't add test for `test_terminal_collision` because I'm not exactly sure what is happening with this example with lexical disambiguation turned on. And result is not something I consider intuitive. (That's why I opened issue #84.)

Instead I added two tests: one testing preferring longer terminal matches and one preferring terminal match over STOP token.